### PR TITLE
 #415 Ensure data access is compatible with JDK17

### PR DESCRIPTION
**Pull request by cwoods-cpointe**
Update kafka client version in data access

### CODE REVIEW

**Review by carter-cundiff — APPROVED** [review]

File: foundation/foundation-mda/src/main/resources/templates/data-access/data.access.pom.xml.vm [hunk 1]
```diff
@@ -92,11 +92,10 @@
             <artifactId>stax</artifactId>
             <version>1.2.0</version>
         </dependency>
-        <!-- vertx-kafka-client currently bound to 2.x versions -->
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>kafka-clients</artifactId>
-            <version>2.8.1</version>
+            <version>${version.kafka.client}</version>
```
Comment (carter-cundiff) [inline foundation/foundation-mda/src/main/resources/templates/data-access/data.access.pom.xml.vm:98]: This template change will be handled by a baton migration in my ticket (#351)

File: build-parent/pom.xml [hunk 2]
```diff
@@ -67,6 +67,7 @@
         <version.awaitility>4.0.3</version.awaitility>
         <version.plexus.util>3.5.1</version.plexus.util>
         <version.jackson.mapper.asl>1.9.3</version.jackson.mapper.asl>
+        <version.kafka.client>3.8.0</version.kafka.client>
```
Comment (ewilkins-csi) [inline build-parent/pom.xml:70]: Q: Do we need to also update the kafka chart version, or is the client compatible?

Comment (cwoods-cpointe) [reply build-parent/pom.xml:70]: Carter was able to test wit 3.8 and it is compatible. However I noticed data-access module doesnt need to use this version and exclude the quarkus version any more. So will remove it here and remove the exclusion from the quarkus dependency

Comment (carter-cundiff) [reply build-parent/pom.xml:70]: Thinking we should add something to the release notes about how that exclusion is no longer necessary

Comment (cwoods-cpointe) [reply build-parent/pom.xml]: Added a blurb in the release notes

**Pull request merged**